WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSTabEligibilityChecker.swift

```diff
@@ -239,28 +239,14 @@ private extension POSTabEligibilityChecker {
     }
 
     func waitForSiteSettingsRefresh() async -> [SiteSetting] {
-        // Uses a shared task so that multiple calls can await the same result since site settings can be emitted only once.
-        if let existingTask = siteSettingsTask {
-            return await existingTask.value
-        }
-
-        let task = Task<[SiteSetting], Never> { [weak self] in
-            guard let self else { return [] }
-
-            for await siteSettings in siteSettings.settingsStream {
-                guard siteSettings.siteID == self.siteID, siteSettings.settings.isNotEmpty, siteSettings.source != .initialLoad else {
-                    continue
-                }
-                siteSettingsTask = nil
-                return siteSettings.settings
+        for await siteSettings in siteSettings.settingsStream.values {
+            guard siteSettings.siteID == siteID, siteSettings.settings.isNotEmpty, siteSettings.source != .initialLoad else {
+                continue
             }
-            // If we get here, the stream completed without yielding any values for our site ID which is unexpected.
-            siteSettingsTask = nil
-            return []
+            return siteSettings.settings
         }
-
-        siteSettingsTask = task
-        return await task.value
+        // If we get here, the stream completed without yielding any values for our site ID which is unexpected.
+        return []
```

Comment:
Should we throw an error here with something like `.ineligible(reason: .siteSettingsNotAvailable)` ? Otherwise we're not distinguishing between legitimate empty settings (which should never happen?) vs stream failed.

Reply:
Good question - if an empty array is returned from here, `.ineligible(reason: .siteSettingsNotAvailable` is returned in the call site:

https://github.com/woocommerce/woocommerce-ios/blob/4e7020622c7fe8ca65e7ae8daef9e9923cbb9851/WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSTabEligibilityChecker.swift#L229-L232